.github/workflows/build-test-publish.yml

```
@@ -202,35 +206,66 @@ jobs:
               | jq --exit-status -r '.raw_id_token'
           )
           echo "${REGISTRY_TOKEN}" | docker login '${{ secrets.REGISTRY_HOST }}' -u '${{ secrets.REGISTRY_USER }}' --password-stdin
+      - name: Configure AWS credentials
+        if: matrix.tag_ecr_public != ''
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/${{ secrets.AWS_ECR_ROLE }}
+          aws-region: ${{ vars.AWS_REGION }}
+      - name: Log in to Amazon ECR Public
+        if: matrix.tag_ecr_public != ''
+        id: login-ecr-public
+        uses: aws-actions/amazon-ecr-login@v2
+        with:
+          registry-type: public
       - name: Tag builder and push to Docker Hub
-        if: matrix.tag_public != ''
+        if: matrix.tag_docker_hub != ''
         run: |
-          PUBLIC_IMAGE_URI='${{ matrix.tag_public }}'
+          DOCKER_HUB_IMAGE_URI='${{ matrix.tag_docker_hub }}'
           set -x
-          docker tag '${{ matrix.builder }}' "${PUBLIC_IMAGE_URI}"
-          docker push "${PUBLIC_IMAGE_URI}"
+          docker tag '${{ matrix.builder }}' "${DOCKER_HUB_IMAGE_URI}"
+          docker push "${DOCKER_HUB_IMAGE_URI}"
       - name: Tag builder and push to internal registry
         if: matrix.tag_private != ''
         run: |
           PRIVATE_IMAGE_URI='${{ secrets.REGISTRY_HOST }}/s/${{ secrets.SERVICE_TOKEN_USER_NAME }}/${{ matrix.tag_private }}'
           set -x
           docker tag '${{ matrix.builder }}' "${PRIVATE_IMAGE_URI}"
           docker push "${PRIVATE_IMAGE_URI}"
+      - name: Tag builder and push to public ECR
+        if: matrix.tag_ecr_public != ''
+        run: |
+          ECR_PUBLIC_IMAGE_URI='public.ecr.aws/${{ matrix.tag_ecr_public }}'
+          set -x
+          docker tag '${{ matrix.builder }}' "${ECR_PUBLIC_IMAGE_URI}"
+          docker push "${ECR_PUBLIC_IMAGE_URI}"
 
-  publish-manifest:
+  publish-manifests:
```

For the other job names we're using the singular even if there is a matrix, since each job only does a single action:

```
create (builder-24, arm64)
test (go, builder-24, arm64)
publish-image (builder-20, amd64, heroku/builder:20)
```

ie: The `publish-image (builder-20, amd64, heroku/builder:20)` job only publishes one local image.

```suggestion
  publish-manifest:
```

---

Ah yeah, I had a previous iteration where it was pushing multiple per job.